starlite/plugins/base.py

```diff
@@ -54,3 +54,20 @@ def get_plugin_for_value(value: Any, plugins: List[PluginProtocol]) -> Optional[
 class PluginMapping(NamedTuple):
     plugin: PluginProtocol
     model_class: Any
+
+    def get_value_converted_to_model_class(
+        self, value: Union[BaseModel, List[BaseModel], Tuple[BaseModel, ...]]
+    ) -> Any:
+        """
+        Given some generated value by plugin,
+        return instance of original class.
+
+        Also excepts can accept list or tuple of values.
+        """
+
+        def get_instance(value: BaseModel) -> Any:
+            return self.plugin.from_pydantic_model_instance(self.model_class, pydantic_model_instance=value)
+
+        if isinstance(value, (list, tuple)):
```

Comment:
Not sure if we have a util already, you can create a util to check if value is a non string iterable. 

Optional.

Reply:
I'm not sure it is possible from typing perspective. This won't work with if block:

```python
@overload
def is_nonstring_iterable(value: list[T]) -> TypeGuard[list[T]]:
    ...


@overload
def is_nonstring_iterable(value: tuple[T, ...]) -> TypeGuard[tuple[T, ...]]:
    ...


@overload
def is_nonstring_iterable(value: Any) -> Any:
    ...


def is_nonstring_iterable(value: Any | list[T] | tuple[T, ...]) -> Any | TypeGuard[list[T]] | TypeGuard[tuple[T, ...]]:
    return isinstance(value, (list, tuple))

```

Reply:
Use the `Sequence` type?

Reply:
`str` is `Sequence[str]` and the function is named `is_nonstring_iterable` 🤔

Reply:
Nvm, not worth to waste time on.

Reply:
I agree.